parsl/executors/status_handling.py

```diff
@@ -44,7 +44,7 @@ class BlockProviderExecutor(ParslExecutor):
     I'm not sure of use cases for switchability at the moment beyond "yes or no"
     """
     def __init__(self, *,
-                 provider: ExecutionProvider,
+                 provider: Optional[ExecutionProvider],
                  block_error_handler: bool):
```

Comment:
If this is optional per provider, do the individual providers need an additional warning check when it is provided and not necessary, or not provided and required?